stereo_points/stereo_points.py

```diff
@@ -107,205 +104,194 @@ def setup(self):
+        self.fiducialGroup_selectionCombo.connect("currentNodeChanged(vtkMRMLNode*)", self.onControlPointSelectedChanged)
+        self.disorient_btn.connect("clicked(bool)", self.onDisorientBtnClicked)
 
-        self.addBtn.connect('clicked(bool)', self.onAddBtnClicked)
+        self.addBtn.connect("clicked(bool)", self.onAddBtnClicked)
 
-        self.fiducialNodesToCoordTables_id_tuples = list()
-
-    ###################################################################################################
+    #########################################################################################################
     # connections
-
     #########################################################################################################
     def onReferenceImageSelectedChanged(self, newNode):
-        # coordTable = slicer.mrmlScene.GetNodesByName(self.fiducialGroup_selectionCombo.currentNode().GetName() + "_coordsConversion").GetItemAsObject(0)
-        coordTable = slicer.mrmlScene.GetNodeByID(
-            self.findMatchingNodeInIdTupleList(self.fiducialNodesToCoordTables_id_tuples,
-                                               self.fiducialGroup_selectionCombo.currentNode().GetID()))
-        self.updatePointsCoordsFromXYZ(coordTable, newNode, self.frameTransform_selectionCombo.currentNode())
-        self.disorient_btn.setText('Disorient "' + newNode.GetName() + '"')
+        # we only update the table if one is selected.
+        if newNode is None:
+            return
+        if self.GetCoordTable():
+            self.updatePointsCoordsFromXYZ(self.GetCoordTable(), newNode, self.frameTransform_selectionCombo.currentNode())
+            self.disorient_btn.setText('Disorient "' + newNode.GetName() + '"')
 
     def onFrameTransformSelectedChanged(self, newNode):
-        # coordTable = slicer.mrmlScene.GetNodesByName(self.fiducialGroup_selectionCombo.currentNode().GetName() + "_coordsConversion").GetItemAsObject(0)
-        coordTable = slicer.mrmlScene.GetNodeByID(
-            self.findMatchingNodeInIdTupleList(self.fiducialNodesToCoordTables_id_tuples,
-                                               self.fiducialGroup_selectionCombo.currentNode().GetID()))
-        self.updatePointsCoordsFromXYZ(coordTable, self.referenceImage_selectionCombo.currentNode(), newNode)
+        if (newNode is not None) and (self.GetCoordTable() is not None):  # to not run the updatepoints when nothing is selected
+            self.updatePointsCoordsFromXYZ(self.GetCoordTable(), self.referenceImage_selectionCombo.currentNode(), newNode)
 
     def onControlPointSelectedChanged(self, newNode):
-        # print('selection changed !')
-        if (type(newNode) == type(slicer.vtkMRMLMarkupsLineNode())):
+        # for each markupLineNode, we maintain a table with the coord conversion.
+        # table nodes are names with the name of the markupLine with _coordsConversion as prefix
+        if type(newNode) == type(slicer.vtkMRMLMarkupsLineNode()):
```

Comment:
`type(slicer.vtkMRMLMarkupsLineNode())` is unnecessary and it is an expensive call, as it instantiates an object, just to get its type. You can simply use the type instead:

```suggestion
        if type(newNode) == slicer.vtkMRMLMarkupsLineNode:
```